cpp/src/arrow/compute/exec/util.h

```diff
@@ -234,6 +242,8 @@ class bit_util {
 }  // namespace util
 namespace compute {
 
+class ExecNode;
```

Comment:
Instead can we include `arrow/compute/type_fwd.h`?

Reply:
Looks like the header is already included and the forward declaration here can be removed.